### PR TITLE
Passwort reset UI

### PR DESCRIPTION
**Pull request by gnunicorn**
Passwort reset flow UI implemented. Superseeding #1675 . Includes a e2e UI test for the new flow:

https://github.com/acterglobal/a3/assets/40496/ee646ac9-b803-4fa2-abd0-6bfac86d5cfa

### CODE REVIEW

**Review by kumarpalsinh25 — APPROVED** [review]
Flutter side changes looks good to me.

File: native/acter/src/api/auth.rs [hunk 1]
```diff
@@ -545,34 +545,38 @@ pub async fn reset_password(
...
-    }
+            if resp.status() != StatusCode::OK {
+                let text = resp.text().await?;
+                bail!("reset_password failed: {}", text);
+            }
 
-    Ok(true)
+            Ok(true)
+        })
+        .await?
```
Comment (bitfriend) [inline native/acter/src/api/auth.rs:581]: It is great job that you wrapped `reset_password()` by `RUNTIME.spawn()`.
It is needed before `reset_password()` can be exposed to flutter side.

**Review by bitfriend — APPROVED** [review]
Looks good to me

File: native/acter/src/api/auth.rs (hunk 1 above)
Comment (gnunicorn) [reply native/acter/src/api/auth.rs:581]: yeah and it crashed when I tried using it from the flutter side :D ...

**Review by gtalha07 — APPROVED** [review]
LGTM!

**Pull request merged**